lib/slack-ruby-bot/rspec/support/slack-ruby-bot/respond_with_slack_messages.rb

```diff
@@ -0,0 +1,34 @@
+require 'rspec/expectations'
+RSpec::Matchers.define :respond_with_slack_messages do |expected|
+  include SlackRubyBot::SpecHelpers
+  match do |actual|
+    raise ArgumentError, 'respond_with_slack_messages expects an array of ordered responses' unless expected.respond_to? :each
+    client = respond_to?(:client) ? send(:client) : SlackRubyBot::Client.new
+    def client.test_messages
+      @test_received_messages
+    end
+
+    def client.say(options = {})
+      super
+      @test_received_messages = @test_received_messages.nil? ? '' : @test_received_messages
+      @test_received_messages += "#{options.inspect}\n"
+    end
+
+    message_command = SlackRubyBot::Hooks::Message.new
+    channel, user, message = parse(actual)
+
+    allow(Giphy).to receive(:random) if defined?(Giphy)
+
+    allow(client).to receive(:message)
+    message_command.call(client, Hashie::Mash.new(text: message, channel: channel, user: user))
+    @messages = client.test_messages
+    expected.each do |exp|
```

Comment:
If you make this `expect(...)` above you don't need to do this whole saving of `@messages` and `failure_message`, no? This seems unnecessarily complicated.

Reply:
The expect statements are still there, the issue is that when using the expect statement without the custom failure message and the storage of the received messages, you don't get any useful output. 

#### This is the output without the custom failure message:
 1) RSpec it_fails_on_purpose_to_show_output
     Failure/Error:
       expect(message: "#{SlackRubyBot.config.user} single message")
         .to respond_with_slack_message(/failure message/i)
     
       `expected {:message=>"rubybot single message"} to respond with slack message /failure message/i`
 
  2) RSpec respond_with_failure_to_show_output
     Failure/Error:
       expect(message: "#{SlackRubyBot.config.user} respond 5 times")
         .to respond_with_slack_messages(expected_responses)
     
       `expected {:message=>"rubybot respond 5 times"} to respond with slack messages /response 0/i, /response 1/i, /response 2/i, /response 3/i, /response 4/i, and /response 7/i`

#### This is the output with the custom failure message, much more useful in figuring out why it failed.     
1) RSpec it_fails_on_purpose_to_show_output
     Failure/Error:
       expect(message: "#{SlackRubyBot.config.user} single message")
         .to respond_with_slack_message(/failure message/i)
     
       `expected to receive message with text: (?i-mx:failure message) once,
        received:[{:text=>"single response", :channel=>"channel"}]`

  2) RSpec respond_with_failure_to_show_output
     Failure/Error:
       expect(message: "#{SlackRubyBot.config.user} respond 5 times")
         .to respond_with_slack_messages(expected_responses)
     
       `expected to receive messages in order with text: [/response 0/i, /response 1/i, /response 2/i, /response 3/i, /response 4/i, /response 7/i]
        received:[{:text=>"response 0", :channel=>"channel"}, {:text=>"response 1", :channel=>"channel"}, {:text=>"response 2", :channel=>"channel"}, {:text=>"response 3", :channel=>"channel"}, {:text=>"response 4", :channel=>"channel"}]`